storage/heap/ha_heap.cc

```diff
@@ -856,15 +856,6 @@ heap_prepare_hp_create_info(TABLE *table_arg, bool internal_table,
       {
         fixed_data_size= next_field_pos;
       }
-
```

Comment:
Deleting this code effectively makes fixed_key_fieldnr local var a constant zero, and then hp_create_info->fieked_key_fieldnr initialisation at the bottom of this function is always initialized to 0 too. Looking at how this field is used in heap_create, by code review, this would result in share->fixed_column_count / fixed_data_length to be initialized with too large values - can you please check?

Reply:
No, if we consider creating and altering table T1 from my addition to heap.test, then share->fixed_column_count=2 and fixed_data_length=16 for both creating and altering the table. That seems reasonable. While debugging heap_create without the patch exposes fixed_column_count=4 and fixed_data_length=30 when altering the table, what is obviously wrong. 
Actually I could not quite figure out the meaning of field fixed_key_fieldnr and even suppose it's correct value for T1 case (1, 2 or smth else?) . Deleting code really makes it constant 0, so we may consider removing this var from code at all.  
It seems to me that looping through all columns in heap_create (line 136) starting from 0 should give us correct results, so is there any need in this field?

Reply:
Indeed, I agree that loop at heap_create can just start from zero and that fixed_key_fieldnr can be simply deleted

Reply:
Deleted fixed_key_fieldnr but still could not handle those submodules to make the patch mergeable, sorry.